**Data Pre-Processing**

***ENCODING FOR Categorical columns***

In [18]:
import pandas as pd
from sklearn.preprocessing import LabelEncoder

# Assuming 'df' is your DataFrame
label_encoder = LabelEncoder()
data['Credentials of the Provider'] = label_encoder.fit_transform(data['Credentials of the Provider'])
data['Credentials of the Provider']

0         667
1         667
2         438
3         956
4         410
         ... 
99995    1594
99996     145
99997     667
99998    1854
99999    1854
Name: Credentials of the Provider, Length: 100000, dtype: int32

The original data had several different credentials like "MD", "DO", "RN", etc., the LabelEncoder would assign a unique integer to each unique credential.
The number 667 corresponds to one of the credentials, and the other numbers like 438, 956, etc., correspond to different credentials.

In [19]:
data

index  National Provider Identifier  \
0      8774979                    1891106191   
1      3354385                    1346202256   
2      3001884                    1306820956   
3      7594822                    1770523540   
4       746159                    1073627758   
...        ...                           ...   
99995  3837311                    1386938868   
99996  2079360                    1215091327   
99997  8927965                    1902868185   
99998  8854571                    1891941183   
99999  3547535                    1356772156   

      Last Name/Organization Name of the Provider First Name of the Provider  \
0                                     UPADHYAYULA                  SATYASREE   
1                                           JONES                      WENDY   
2                                        DUROCHER                    RICHARD   
3                                         FULLARD                     JASPER   
4                                        PERROTTI                    ANTHONY   
...                                           ...                        ...   
99995                                       PAPES                       JOAN   
99996                                      HAYNER                   MARGARET   
99997                                    VALENCIA                       DANA   
99998                              GONZALEZ-LAMOS                    RAFAELA   
99999                                    RAMEZANI                     ELIIAN   

       Credentials of the Provider Gender of the Provider  \
0                              667                      F   
1                              667                      F   
2                              438                      M   
3                              956                      M   
4                              410                      M   
...                            ...                    ...   
99995                         1594                      F   
99996                          145                      F   
99997                          667                      M   
99998                         1854                      F   
99999                         1854                      F   

      Entity Type of the Provider City of the Provider  \
0                               I          SAINT LOUIS   
1                               I         FAYETTEVILLE   
2                               I          NORTH HAVEN   
3                               I          KANSAS CITY   
4                               I              JUPITER   
...                           ...                  ...   
99995                           I           WILMINGTON   
99996                           I              REDMOND   
99997                           I          SAINT LOUIS   
99998                           I            LARCHMONT   
99999                           I           GREAT NECK   

       Zip Code of the Provider State Code of the Provider  ... HCPCS Code  \
0                   631041004.0                         MO  ...      99223   
1                   283043815.0                         NC  ...      G0202   
2                    64732343.0                         CT  ...      99348   
3                   641183998.0                         MO  ...      81002   
4                   334585700.0                         FL  ...      96372   
...                         ...                        ...  ...        ...   
99995               604811236.0                         IL  ...      97162   
99996               977561502.0                         OR  ...      99213   
99997               631312322.0                         MO  ...      93320   
99998               105383500.0                         NY  ...      G0008   
99999                   11023.0                         NY  ...      97112   

                                       HCPCS Description HCPCS Drug Indicator  \
0      Initial hospital

In [20]:
# Import necessary libraries
import pandas as pd
from sklearn.preprocessing import LabelEncoder
# Initialize Label Encoders
gender_encoder = LabelEncoder()
entity_type_encoder = LabelEncoder()
city_encoder = LabelEncoder()
drug_indicator_encoder = LabelEncoder()
# Apply Label Encoding to the columns
data['Gender of the Provider'] = gender_encoder.fit_transform(data['Gender of the Provider'])
data['Entity Type of the Provider'] = entity_type_encoder.fit_transform(data['Entity Type of the Provider'])
data['City of the Provider'] = city_encoder.fit_transform(data['City of the Provider'])
data['HCPCS Drug Indicator'] = drug_indicator_encoder.fit_transform(data['HCPCS Drug Indicator'])


# Display the updated DataFrame
print(data)


         index  National Provider Identifier  \
0      8774979                    1891106191   
1      3354385                    1346202256   
2      3001884                    1306820956   
3      7594822                    1770523540   
4       746159                    1073627758   
...        ...                           ...   
99995  3837311                    1386938868   
99996  2079360                    1215091327   
99997  8927965                    1902868185   
99998  8854571                    1891941183   
99999  3547535                    1356772156   

      Last Name/Organization Name of the Provider First Name of the Provider  \
0                                     UPADHYAYULA                  SATYASREE   
1                                           JONES                      WENDY   
2                                        DUROCHER                    RICHARD   
3                                         FULLARD                     JASPER   
4                      